src/parser/rogue/subtlety/modules/features/VanishFindWeakness.tsx

```diff
@@ -0,0 +1,152 @@
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent} from 'parser/core/Events';
+import SPELLS from 'common/SPELLS';
+import React from 'react';
+import { NumberThreshold, ThresholdStyle, When } from 'parser/core/ParseResults';
+import { t } from '@lingui/macro';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import SpellLink from 'common/SpellLink';
+import { formatMilliseconds, formatNumber } from 'common/format';
+import BoringSpellValue from 'interface/statistics/components/BoringSpellValue';
+import Enemies from 'parser/shared/modules/Enemies';
+import Enemy from 'parser/core/Enemy';
+
+/**
+ * Analyzer for the suggestion that Vanish should not be cast
+ * with longer than 1s of Find Weakness buff remaining. This is because
+ * Vanish allows the rogue to re-cast Shadowstrike and re-apply Find Weakness.
+ */
+class VanishFindWeakness extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+  }
+
+  private BAD_CAST_WINDOW: number = 1000;
+  private FW_DEBUFF_LENGTH: number = 18 * 1000;
+  private DEBUFF_CAP: number = this.FW_DEBUFF_LENGTH; // Verified with Ravenholdt discord that FW does not follow pandemic debuff length rules (1.3 times base length).
+
+  private lastVanishPtr: CastEvent | null = null;
+  private badVanishCasts: Map<CastEvent, number> = new Map<CastEvent, number>();
+  protected enemies!: Enemies;
+
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.VANISH), this.onVanishCast);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SHADOWSTRIKE), this.onFindWeaknessApplierCast);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.CHEAP_SHOT), this.onFindWeaknessApplierCast);
+  }
+
+  onVanishCast(event: CastEvent): void {
+      this.lastVanishPtr = event;
+  }
+
+  onFindWeaknessApplierCast(event: CastEvent): void {
+    if (event.targetID === undefined) {
+      this.debug(`Skipping FW buff tracking for FW applier cast at timestamp ${event.timestamp} due to missing target ID.`);
```

Comment:
This will print in prod which is a nono

Reply:
Got it. I will remove the debug printing through this PR.